Neos.Fusion/Classes/FusionObjects/DataStructureImplementation.php

```diff
@@ -96,6 +98,21 @@ private function isUntypedProperty($property): bool
         if (!is_array($property)) {
             return false;
         }
-        return array_intersect_key(array_flip(Parser::$reservedParseTreeKeys), $property) === [];
+
+        $keysToIdentifyTypedProperty = array_flip(Parser::$reservedParseTreeKeys);
```

Comment:
`$reservedParseTreeKeys = ['__meta', '__prototypes', '__stopInheritanceChain', '__prototypeObjectName', '__prototypeChain', '__value', '__objectType', '__eelExpression'];`

has to much stuff - for this feature we only need to check for:
one of these: `['__value', '__objectType', '__eelExpression']` - otherwise we are adding buggy functionality as we have see this issue ^^

should these 3 'magic' keys be inline in the `DataStructureImplementation` or be exposed seperately by the parser?

Reply:
I would hardcode those now in the DataStructureImplementation.

Reply:
BTW: Found this in fusion forms https://github.com/neos/fusion-form/blob/master/Classes/Runtime/FusionObjects/AbstractCollectionFusionObject.php#L130